internal/clients/http/gcp/gcp_client.go

```diff
@@ -0,0 +1,92 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/aws/smithy-go/ptr"
+	"github.com/rs/zerolog"
+)
+
+type gcpClient struct {
+	instancesClient *compute.InstancesClient
+	context         context.Context
+	logger          zerolog.Logger
+}
+
+func init() {
+	clients.GetGCPClient = newGCPClient
+}
+
+func newGCPClient(ctx context.Context) (clients.GCP, error) {
+	logger := ctxval.Logger(ctx).With().Str("client", "gcp").Logger()
+
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create new instances REST client: %w", err)
+	}
+
+	return &gcpClient{
+		instancesClient: instancesClient,
+		context:         ctx,
+		logger:          logger,
+	}, nil
+
+}
+
+func (c *gcpClient) Close() {
+	defer c.instancesClient.Close()
```

Comment:
defer makes the statement run at the end of the function, so in this case it is really not needed, given we have this statement at the end of the function ^_^

Reply:
I don't have it at the end of the function

Reply:
it's the only statement in the function, so you do have it at the end :)

Reply:
Yeah, you simply call `Close()` in this function and then you use `defer` at the call-site of this function. In other words, move `defer` from here to where you call this function.